src/rust/engine/build_utils/src/lib.rs

```diff
@@ -0,0 +1,143 @@
+use std::env;
+use std::ffi::OsStr;
+use std::io;
+use std::ops::Deref;
+use std::path::PathBuf;
+use std::process::{Command, Output};
+
+#[derive(Clone, Debug)]
+pub struct ExecutionError {
+  status: Option<i32>,
+  stdout: String,
+  stderr: String,
+}
+
+fn to_string(data: Vec<u8>) -> String {
+  String::from_utf8(data).unwrap()
+}
+
+impl ExecutionError {
```

Comment:
`impl From<Output> for ExecutionError` feels a little more idiomatic than a random method named `from` (and means that below you would just `Err(output.into())`)

Reply:
Agreed, fixed.